### PR TITLE
feat: add azure-director

### PR DESCRIPTION
**Pull request by dspo**
#### What this PR does / why we need it:
feat: add azure-director

- 增加插件 azure-director
- 增加 credentials: api-key 分发功能, 及其 admin 接口, 以及基于 api-key 的认证
- 根据 api-key 和 request header 决定 direct  的实例, 而不仅仅是 route.Router 配置 (route.Router 支持暂时下掉了, 毕竟我们不是要做一个完备的反向代理中间件或网关)

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | feat: add azure-director             |
| 🇨🇳 中文    | 增加 azure-director 插件             |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/proto_fetcher.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
 EXTERNAL_REPO_DIR="externalrepo"
 
 # otel remote proto conf
-OTEL_PROTO_REPO="https://github.com/open-telemetry/opentelemetry-proto.git"
+OTEL_PROTO_REPO="https://ghproxy.com/https://github.com/open-telemetry/opentelemetry-proto.git"
```
Comment (dspo) [inline api/proto/proto_fetcher.sh:9]: @sfwn 是否可以修改 ?

File: .erda/ai-proxy/migrations/ai-proxy/20230425-filter-audit.sql [hunk 2]
```diff
@@ -5,6 +5,8 @@ CREATE TABLE `ai_proxy_filter_audit`
     `updated_at`            DATETIME     NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '更新时间',
     `deleted_at`            DATETIME     NOT NULL DEFAULT '1970-01-01 00:00:00' COMMENT '删除时间, 1970-01-01 00:00:00 表示未删除',
 
+    `app_key_sha256`        CHAR(64)     NOT NULL DEFAULT '' COMMENT '请求使用的 app_key sha256 哈希值',
```
Comment (sfwn) [inline .erda/ai-proxy/migrations/ai-proxy/20230425-filter-audit.sql:8]: what is app_key?

File: .erda/ai-proxy/migrations/ai-proxy/20230706-credentials.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    `access_key_id`     CHAR(32)     NOT NULL COMMENT '平台接入 AI 服务的 AK',
+    `secret_key_id`     CHAR(32)     NOT NULL COMMENT '平台接入 AI 服务的 SK',
+    `name`              VARCHAR(64)  NOT NULL COMMENT '凭证名称',
+    `platform`          VARCHAR(128) NOT NULL COMMENT '接入 AI 服务的平台',
+    `description`       VARCHAR(512) NOT NULL COMMENT '凭证描述',
+    `enabled`           BOOLEAN      NOT NULL DEFAULT true COMMENT '是否启用该凭证',
+    `expired_at`        DATETIME     NOT NULL DEFAULT '2099-01-01 00:00:00' COMMENT '凭证过期时间',
+
+    `provider_name`     VARCHAR(64)  NOT NULL DEFAULT '' COMMENT 'AI 服务 Provider 名称',
```
Comment (sfwn) [inline .erda/ai-proxy/migrations/ai-proxy/20230706-credentials.sql:16]: enlarge field size

Comment (sfwn) [reply .erda/ai-proxy/migrations/ai-proxy/20230706-credentials.sql:16]: add examples in comment

File: .erda/ai-proxy/migrations/ai-proxy/20230706-credentials.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    `provider_instance` VARCHAR(64)  NOT NULL DEFAULT '' COMMENT 'AI 服务 Provider 实例 id',
```
Comment (sfwn) [inline .erda/ai-proxy/migrations/ai-proxy/20230706-credentials.sql:17]: `provider_instance_id`

File: api/proto/apps/aiproxy/provider.proto [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      post: "/api/ai-proxy/providers"
+    };
+    option(erda.common.openapi) = {
+      path: "/api/ai-proxy/providers"
+    };
+  }
+
+  rpc DeleteProvider(Provider) returns (common.VoidResponse) {
+    option(google.api.http) = {
+      delete: "/api/ai-proxy/providers/{name}/{instanceId}"
```
Comment (sfwn) [inline api/proto/apps/aiproxy/provider.proto:31]: /api/ai-proxy/providers/{name}/instances/{instanceId}

File: api/proto/proto_fetcher.sh (hunk 1 above)
Comment (sfwn) [reply api/proto/proto_fetcher.sh:9]: you'd better not.

File: cmd/ai-proxy/conf/providers.yml [hunk 6]
```diff
@@ -25,10 +26,22 @@ providers:
     scheme: https
     description: azure 提供的 ai 能力
     docSite: https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference
+    appKey: ${ env.OPENAI_API_KEY }
```
Comment (sfwn) [inline cmd/ai-proxy/conf/providers.yml:29]: bad env name

File: cmd/ai-proxy/conf/providers.yml [hunk 7]
```diff
@@ -25,10 +26,22 @@ providers:
...
     metadata:
       RESOURCE_NAME: "terminus3"
       DEVELOPMENT_NAME: "gpt-35-turbo-0301"
 
+  - name: azure
+    instanceId: terminus2
```
Comment (sfwn) [inline cmd/ai-proxy/conf/providers.yml:35]: instanceId should be composed with resource_name & deployment_name

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/ai-proxy/conf/providers.yml [hunk 8]
```diff
@@ -15,6 +15,7 @@ providers:
     scheme: https
```
Comment (dspo) [inline cmd/ai-proxy/conf/providers.yml:15]: remove default

**Review by sixther-dc — APPROVED** [review]
/approve

**Review by sixther-dc — APPROVED** [review]
/approve

**Review by sfwn — APPROVED** [review]

**Pull request merged**